### PR TITLE
Up cucumber-jvm version to 4.0.0 (Fixes #2)

### LINKED ISSUE #2: Update cucumber-jvm version
The archetype is at 2.3.1 while cucumber-jvm is (currently) at 4.0.0

### PR DESCRIPTION
**Pull request by mlvandijk**
* Up the cucumber-jvm version from 2.3.1 to 4.0.0 (current version)
* Use one version for cucumber-jvm in pom.xml (easier updates/less trouble)
* Update maven archetype version => in line with cucumber-jvm version (as suggested by @mpkorstanje )

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,5 +6,5 @@ This is a Maven Archetype for setting up an empty Cucumber project.
 mvn archetype:generate                                  \
   -DarchetypeGroupId=io.cucumber                        \
   -DarchetypeArtifactId=cucumber-archetype              \
-  -DarchetypeVersion=2.3.1.2                            \
+  -DarchetypeVersion=2.3.1.3                            \
```
Comment (mpkorstanje) [inline README.md]: Would it be sensible to lock the archetype version to cucumber jvms version?

Comment (mlvandijk) [reply README.md]: Yes, it would. Thanks!

**Comment by mlvandijk** [comment]
@mpkorstanje - If the maven archetype version matches the cucumber-jvm version, should there be an archetype for each jvm version?

**Comment by mpkorstanje** [comment]
I think the easiest way to automate this would be making it a module of cucumber-jvm.
But we'd have to figure out how to test the generated archetype and how to get the current project version into the archetype pom.

**Pull request merged**